### DISCUSSION TITLE
Is it possible to run Orchard Core CMS with a read-only App_Data folder? (usecase: Azure Appservice, with WEBSITE_RUN_FROM_PACKAGE  enabled)

### DISCUSSION THREAD

**Discussion by rikbosch** [OrchardCMS/OrchardCore#14684] [Q&A] [answered]
We're trying to host a multitenant Orchard Core application on Azure Appservice. 

To improve cold startup time, it's adviced to set the `WEBSITE_RUN_FROM_PACKAGE` environment variable to `1`.  (see https://learn.microsoft.com/en-us/azure/app-service/deploy-run-package)

This however makes the entire website `readonly`, which causes a problem, especially with the `OC.Media` module. We're configuring Azure Storage as a Media provider, but it looks like OC is still trying to cache certain files locally.

I can't find anything about this behavior, but our site fails to startup because it's trying to create a directory inside the App_Data folder.

If by default Orchard Core is caching static assets locally, that would not make sense when it's running inside Azure AppService, because the local filesystem itself is hosted on Azure Storage. What would we the best possible way to remove the media caching behavior?

**Comment by hishamco** [comment]
> Is it possible to run Orchard Core CMS with a read-only App_Data folder?

I don't think so, the tenant's info, media, caching, and many other things are using `App_Data`, the question is why do you want it to be read-only?!!

**Reply by rikbosch** [reply]
I want the server to be stateless, all state should be in azure storage. We're running our application in an Azure App Service, all storage operations are already performed on azure storage underneath. Azure App Service shares the application files with every instance ( see https://github.com/projectkudu/kudu/wiki/Understanding-the-Azure-App-Service-file-system ). this can cause locking issues on deployment.

I can map the data folder to a server's local folder using a the [ORCHARD_APP_DATA](https://docs.orchardcore.net/en/latest/docs/reference/core/Configuration/#orchard_app_data-environment-variable) environment variable, but then we risk one tenant clogging up the local file storage of a server, possibly affecting all tenants running on that instance.

We're already loading the tenant info from azure storage, and added the azure media feature as well. It appears that only the local caching feature of orchard is still writing to the APP_DATA folder.

**Comment by ns8482e** [comment, marked as answer]
@rikbosch  App_Data requires read/write access. However  you can achieve this by mounting data directory separate from your application  Use ORCHARD_APP_DATA environment variable and mount  azure storage as local storage instead of local folder https://learn.microsoft.com/en-us/azure/app-service/configure-connect-to-azure-storage?tabs=basic%2Cportal&pivots=container-linux
